src/fs/syslib/kfile.c

```diff
@@ -57,7 +57,7 @@ struct idesc *kopen(struct node *node, int flag) {
 			return NULL;
 		}
 
-		if (S_ISCHR(node->mode)) {
+		if (S_ISCHR(node->mode) || S_ISBLK(node->mode)) {
 			return char_dev_open(node, flag);
```

Comment:
Calling `char_dev_open()` for block device may look like a typo. I think that it's a good idea to write comment here or rename function itself.

Reply:
done.
Add a comment bellow